frontend/lib/src/components/shared/BaseColorPicker/BaseColorPicker.tsx

```diff
@@ -47,113 +49,126 @@ export interface BaseColorPickerProps {
+        userSelect: "none" as const,
+      },
+      body: {
+        padding: theme.spacing.xl,
+      },
+    },
   }
 
-  public render(): React.ReactNode {
-    const { width, showValue, label, labelVisibility, help, disabled } =
-      this.props
-    const { value } = this.state
-
-    return (
-      <StyledColorPicker
-        className="stColorPicker"
-        data-testid="stColorPicker"
-        width={width}
+  return (
+    <StyledColorPicker
+      className="stColorPicker"
+      data-testid="stColorPicker"
+      width={width}
+      disabled={disabled}
+    >
+      <WidgetLabel
+        label={label}
         disabled={disabled}
+        labelVisibility={labelVisibility}
       >
-        <WidgetLabel
-          label={label}
-          disabled={disabled}
-          labelVisibility={labelVisibility}
-        >
-          {help && (
-            <StyledWidgetLabelHelpInline>
-              <TooltipIcon content={help} placement={Placement.TOP_RIGHT} />
-            </StyledWidgetLabelHelpInline>
-          )}
-        </WidgetLabel>
-        <UIPopover
-          onClose={this.onColorClose}
-          content={() => (
-            <StyledChromePicker data-testid="stColorPickerPopover">
-              <ChromePicker
-                color={value}
-                onChange={this.onColorChange}
-                disableAlpha={true}
-              />
-            </StyledChromePicker>
-          )}
-        >
-          <StyledColorPreview disabled={disabled}>
-            <StyledColorBlock
-              data-testid="stColorPickerBlock"
-              backgroundColor={value}
-              disabled={disabled}
+        {help && (
+          <StyledWidgetLabelHelpInline>
+            <TooltipIcon content={help} placement={Placement.TOP_RIGHT} />
+          </StyledWidgetLabelHelpInline>
+        )}
+      </WidgetLabel>
+      <UIPopover
+        key={popoverKey}
+        onClose={onColorClose}
+        placement="bottomLeft"
+        content={() => (
+          <StyledChromePicker data-testid="stColorPickerPopover">
+            <ChromePicker
+              color={value}
+              onChange={onColorChange}
+              disableAlpha={true}
+              styles={customChromePickerStyles}
             />
-            {showValue && (
-              <StyledColorValue>{value.toUpperCase()}</StyledColorValue>
-            )}
-          </StyledColorPreview>
-        </UIPopover>
-      </StyledColorPicker>
-    )
-  }
+          </StyledChromePicker>
+        )}
+      >
+        <StyledColorPreview disabled={disabled}>
+          <StyledColorBlock
+            data-testid="stColorPickerBlock"
+            backgroundColor={value}
+            disabled={disabled}
+          />
+          {showValue && (
+            <StyledColorValue>{value.toUpperCase()}</StyledColorValue>
+          )}
+        </StyledColorPreview>
+      </UIPopover>
+    </StyledColorPicker>
+  )
 }
 
 export default BaseColorPicker
```

Comment:
nit: I think you can also wrap this into a memo (something we are trying to do for all components):

```suggestion
export default memo(BaseColorPicker)
```

Reply:
Done